superduperdb/container/model.py

```diff
@@ -279,87 +360,28 @@ def _predict_with_select_and_ids(
+        if not model_update_kwargs.get('document_embedded', True):
+            collection = f'_outputs.{X}.{self.identifier}'
+
         select.model_update(
             db=db,
             model=self.identifier,
             outputs=outputs,
             key=X,
             ids=ids,
+            collection=collection,
+            flatten=flatten,
         )
         return
 
-    def predict(
-        self,
-        X: t.Any,
-        db: t.Optional[DB] = None,
-        select: t.Optional[Select] = None,
-        distributed: t.Optional[bool] = None,
-        ids: t.Optional[t.List[str]] = None,
-        max_chunk_size: t.Optional[int] = None,
-        dependencies: t.Sequence[Job] = (),
-        listen: bool = False,
-        one: bool = False,
-        context: t.Optional[t.Dict] = None,
-        in_memory: bool = True,
-        overwrite: bool = False,
-        **kwargs,
-    ) -> t.Any:
-        if one:
-            assert db is None, 'db must be None when ``one=True`` (direct call)'
-
-        if isinstance(select, dict):
-            select = Serializable.deserialize(select)
-
-        if db is not None:
-            logging.info(f'Adding model {self.identifier} to db')
-            assert isinstance(self, Component)
-            db.add(self)
-            logging.info('Done.')
-
-        if distributed is None:
-            distributed = s.CFG.cluster.distributed
-
-        if listen:
-            assert db is not None
-            assert select is not None
-            return self._predict_and_listen(X=X, db=db, select=select, **kwargs)
 
-        if distributed:
-            return self.create_predict_job(
-                X,
-                select=select,
-                ids=ids,
-                max_chunk_size=max_chunk_size,
-                overwrite=overwrite,
-                **kwargs,
-            )(db=db, distributed=distributed, dependencies=dependencies)
-        else:
-            if select is not None and ids is None:
-                assert db is not None
-                return self._predict_with_select(
-                    X=X,
-                    select=select,
-                    db=db,
-                    in_memory=in_memory,
-                    max_chunk_size=max_chunk_size,
-                    overwrite=overwrite,
-                    **kwargs,
-                )
-            elif select is not None and ids is not None:
-                assert db is not None
-                return self._predict_with_select_and_ids(
-                    X=X,
-                    select=select,
-                    ids=ids,
-                    db=db,
-                    max_chunk_size=max_chunk_size,
-                    in_memory=in_memory,
-                    **kwargs,
-                )
-            else:
-                if self.takes_context:
-                    kwargs['context'] = context
-                return self._predict(X, one=one, **kwargs)
+@dc.dataclass
+class DefaultModelUpdateKwargs(dict):
```

Comment:
This class not needed (defaults are handled by `model_update` function).

Reply:
so I thought of having a structured class to represent DefaultValues
If we store in model_update the default, its sort of hidden for the developer.

Reply:
I think this is more a documentation thing. Currently this is in the wrong place - we're placing MongoDB only functionality in the base module. We can revisit this later.